### PR TITLE
there is no value need to be deleted when rollback lock

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
There is no value need to be deleted when rollback lock.
@disksing @AndreMouche PTAL

### CODE REVIEW

File: src/storage/mvcc/txn.rs [hunk 1]
```diff
@@ -180,7 +180,8 @@ impl<'a> MvccTxn<'a> {
     pub fn rollback(&mut self, key: &Key) -> Result<()> {
         match try!(self.reader.load_lock(key)) {
             Some(ref lock) if lock.ts == self.start_ts => {
-                if lock.short_value.is_none() {
+                // If prewrite type is DEL or LOCK, it is no need to delete value.
+                if lock.short_value.is_none() && lock.lock_type == LockType::Put {
                     self.delete_value(key, lock.ts);
```
Comment (siddontang) [inline src/storage/mvcc/txn.rs]: any test to cover?

**Comment by disksing** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL  @hhkbp2

**Comment by ngaut** [comment]
Please let @AndreMouche take a look too.

File: src/storage/mvcc/txn.rs (hunk 1 above)
Comment (AndreMouche) [inline src/storage/mvcc/txn.rs]: Is OK since tikv insert into CF-default in prewrite only meet  the condition -- value is long && lock_type is PUT.

**Comment by AndreMouche** [comment]
LGTM

**Pull request merged**